lib/fog/libvirt/models/compute/server.rb

```diff
@@ -360,7 +360,7 @@ def addresses_ip_command(service_arg=service, options={})
           ip_command_global=service_arg.ip_command.nil? ? 'grep $mac /var/log/arpwatch.log|sed -e "s/new station//"|sed -e "s/changed ethernet address//g" |sed -e "s/reused old ethernet //" |tail -1 |cut -d ":" -f 4-| cut -d " " -f 3' : service_arg.ip_command
           ip_command_local=options[:ip_command].nil? ? ip_command_global : options[:ip_command]
 
-          ip_command="mac=#{mac}; server_name=#{name}; "+ip_command_local
+          ip_command="mac=#{mac}; server_name=#{name.gsub(/[\W_-]/, '_')}; "+ip_command_local
```

Comment:
Alright, Just so we don't get lost (Regexp are always complex to read and follow) What do you think of saving this value in a different variable with a more suggested name?

Reply:
Sure I can do that, amended the change.